packages/server/modules/core/migrations/20240710154658_user_emails_backfill.ts

```diff
@@ -0,0 +1,31 @@
+import { Knex } from 'knex'
+import crs from 'crypto-random-string'
+import { scanTableFactory } from '@/modules/core/helpers/scanTable'
+
+export async function up(knex: Knex): Promise<void> {
+  const batchSize = 1000
+
+  for await (const rows of scanTableFactory({ db: knex })({
+    tableName: 'users',
+    batchSize
+  })) {
+    if (rows.length) {
```

Comment:
You should add `debug` level logging here, so in case this gets stuck or is very long, you can at least get info about how many rows have been processed and how many are still pending

Reply:
Also: Maybe you should batch up inserts for improved speed? Just an idea, but maybe batch up around 50 inserts and insert them all at once with one call